### PR TITLE
Update doc

### PR DESCRIPTION
**Pull request by kimlisa**
I might've made crucial modifications, so please review very carefully. (:

### quickstart.md:
The painpoints I ran into when trying to follow the quickstart to a T (locally), was because of who the primary OS user was. The commands in quickstart assumes the primary user will be `teleport`. Problems arise b/c even if we add a linux user named `teleport` with `useradd`, the default primary OS user is still `me`. Even if I map teleport to my OS user, I still get problems because the commands we use on our examples uses `implicit` (default) settings of user. 

So I tried to include instructions to explicitly set users when needed. I think it would've been better all around, if we used explicit commands (explicitly set user), instead of providing shortcuts, but I tried to work with it.

### admin-guide.md:
- added `~/.tsh` section under `Filesystem Layouts`

### CODE REVIEW

File: docs/4.2/admin-guide.md [hunk 1]
```diff
@@ -172,6 +173,11 @@ of them is configurable.
 | `/usr/local/bin/tctl` | Teleport admin tool. It is only needed for auth servers.|
 | `/var/lib/teleport` | Teleport data directory. Nodes keep their keys and certificates there. Auth servers store the audit log and the cluster keys there, but the audit log storage can be further configured via `auth_service` section in the config file.|
 
+#### tsh
+| Full path                 | Purpose                                                                                                                                                                                                                                |
+|---------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `~/.tsh` | Stores temporary SSH certificates, stores user environment variables for reading when `permit_user_env: true`, and cache data such as proxy and user names from last logins. |
```
Comment (kimlisa) [inline docs/4.2/admin-guide.md]: @fspmarshall do you think this is a correct description for `~/.tsh`?

File: docs/4.2/admin-guide.md [hunk 2]
```diff
@@ -51,7 +51,7 @@ $ shasum -a 256 teleport-v{{ teleport.version }}-darwin-amd64-bin.tar.gz
 **Checking Checksum on Linux**
 
 ``` bash
-$ sha256sum teleport-v4.0.8-darwin-amd64-bin.tar.gz
+$ sha256sum teleport-v{{ teleport.version }}-darwin-amd64-bin.tar.gz
```
Comment (benarent) [inline docs/4.2/admin-guide.md]: Good catch

File: docs/4.2/admin-guide.md (hunk 1 above)
Comment (benarent) [reply docs/4.2/admin-guide.md]: I would just reword this to. 

> Temporary store for Proxy Address, Username,  SSH certificates and user environment variables when reading `permit_user_env: trues.

File: docs/4.2/quickstart.md [hunk 3]
```diff
@@ -98,24 +98,31 @@ Congratulations - you are now running Teleport!
...
+    If you do not map to a real OS user you will get authentication errors later on in
     this tutorial!
 
 ``` bash
 # A new Teleport user will be assigned a
 # mapping to an OS user of the same name
 # This is the same as running `tctl users add teleport teleport`
 [teleport@grav-00 ~]$ tctl users add teleport
-Signup token has been created and is valid for 1 hours. Share this URL with the user:
+User teleport has been created but requires a password. Share this URL with the user to complete user setup, link is valid for 1h0m0s:
```
Comment (benarent) [inline docs/4.2/quickstart.md]: It's not a huge deal, but as a rule we should add these updates to a new '5.0' docs, since this update won't be live until then. ( I think it's ok for now, since we're a scrappy team working on docs )

**Review by benarent — CHANGES_REQUESTED** [review]
Looks good, just a few copy changes.

File: docs/4.2/admin-guide.md (hunk 1 above)
Comment (kimlisa) [reply docs/4.2/admin-guide.md]: revised

File: docs/4.2/quickstart.md (hunk 3 above)
Comment (kimlisa) [reply docs/4.2/quickstart.md]: ah that makes sense, I will take that into account next time :)

**Review by benarent — APPROVED** [review]
